src/jestMatchers.ts

```diff
@@ -0,0 +1,338 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import assert from 'assert';
+import type { MetadataBearer } from '@aws-sdk/types';
+import type { AwsCommand, AwsStub } from './awsClientStub';
+
+/**
+ * Provides {@link jest} matcher for testing {@link AwsStub} command calls
+ *
+ * @example
+ *
+ * ```ts
+ * import { mockClient } from "aws-sdk-client-mock";
+ * import { ScanCommand } from "@aws-sdk/client-dynamodb";
+ *
+ * const awsMock = mockClient(DynamoDBClient);
+ *
+ * awsMock.on(ScanCommand).resolves({
+ *   Items: [{ Info: { S: '{ "val": "info" }' }, LockID: { S: "fooId" } }],
+ * });
+ *
+ * it("Should call scan command", async () => {
+ *    // check result ... maybe :)
+ *    await expect(sut()).resolves.toEqual({ ... });
+ *
+ *    // Assert awsMock to have recevied a Scan Command at least one time
+ *    expect(awsMock).toHaveReceivedCommand(ScanCommand);
+ * });
+ * ```
+ */
+export interface AwsSdkJestMockMatchers<E, R> extends Record<string, any> {
+    /**
+     * Asserts {@link AwsStub Aws Client Mock} received a {@link command} exact number of {@link times}
+     *
+     * @param command aws-sdk command constructor
+     * @param times
+     */
+    toHaveReceivedCommandTimes<
+        TCmdInput extends object,
+        TCmdOutput extends MetadataBearer
+    >(
+        command: new (input: TCmdInput) => AwsCommand<TCmdInput, TCmdOutput>,
+        times: number
+    ): R;
+
+    /**
+     * Asserts {@link AwsStub Aws Client Mock} received a {@link command} at least one time
+     *
+     * @param command aws-sdk command constructor
+     */
+    toHaveReceivedCommand<
+        TCmdInput extends object,
+        TCmdOutput extends MetadataBearer
+    >(
+        command: new (input: TCmdInput) => AwsCommand<TCmdInput, TCmdOutput>
+    ): R;
+
+    /**
+     * Asserts {@link AwsStub Aws Client Mock} received a {@link command} at leas one time with input
+     * matching {@link input}
+     *
+     * @param command aws-sdk command constructor
+     * @param input
+     */
+    toHaveReceivedCommandWith<
+        TCmdInput extends object,
+        TCmdOutput extends MetadataBearer
+    >(
+        command: new (input: TCmdInput) => AwsCommand<TCmdInput, TCmdOutput>,
+        input: Partial<TCmdInput>
+    ): R;
+
+    /**
+     * Asserts {@link AwsStub Aws Client Mock} received a {@link command} as defined {@link call} number
+     * with matching {@link input}
+     *
+     * @param call call number to assert
+     * @param command aws-sdk command constructor
+     * @param input
+     */
+    toHaveNthReceivedCommandWith<
+        TCmdInput extends object,
+        TCmdOutput extends MetadataBearer
+    >(
+        call: number,
+        command: new (input: TCmdInput) => AwsCommand<TCmdInput, TCmdOutput>,
+        input: Partial<TCmdInput>
+    ): R;
+}
+
+declare global {
+    // eslint-disable-next-line
```

Comment:
Instead of disabling eslint rule here, please add this to the `.eslintrc.json` under `rules`:

```json
"@typescript-eslint/no-namespace": [
      "error",
      {
        "allowDeclarations": true
      }
    ]
```